python/common/org/python/types/Str.java

```diff
@@ -726,28 +726,40 @@ private long toPositiveIndex(long index) {
         __doc__ = ""
     )
     public org.python.Object isalnum() {
-        throw new org.python.exceptions.NotImplementedError("isalnum() has not been implemented.");
+		if (this.value.matches("(\\p{N}|\\p{L})+")) {
```

Comment:
Hi @SarthakSuri, thanks for contributing !

Can you check what's going on with the indentation here? It looks a bit off, maybe it's mixing tabs and spaces?
Best to stick to spaces-only, as in the rest of the codebase.

About the PR, overall I like where this is going, you seem to be almost done with this, right?
Though I did notice there is a test decorated with `@expectedFailure` still.
Let us know when you think you're done so we can give you more thorough feedback!

You'll also need to sync this with latest codebase, because there are some conflicts now -- ask in Gitter if you need help with this.

Minor open source contributing tip: the easier your PR is to read and understand, the better the chances it has to get merged! :)
It's good to check on GitHub what the PR looks like and try to make the diff as short as possible, minimizing unnecessary changes and explaining when something doesn't look obvious (if a reviewer has to guess why changes are being done, the PR will take a lot more work to review).